src/common/loggingEvents.ts

```diff
@@ -0,0 +1,71 @@
+import { EventType } from './EventType';
+import { PlatformInformation } from './PlatformInformation';
+import { Environment } from '../shared';
+
+export interface BaseEvent {
+    type: EventType;
+}
+
+// Others
+export class EnvironmentChanged implements BaseEvent {
+    type = EventType.EnvironmentChanged;
+    constructor(public env: Environment) { }
+}
+
+// Runtime Dependency
+export class DependencyInstallStart implements BaseEvent {
+    type = EventType.DependencyInstallStart;
+}
+
+export class DependencyInstallFinished implements BaseEvent {
+    type = EventType.DependencyInstallFinished;

```

Comment:
why no constructor for this and the one above?

Reply:
constructor is not nesscery in ts.

---
In reply to: [362393320](https://github.com/docascode/vscode-docs-build/pull/9#discussion_r362393320) [](ancestors = 362393320)